### ISSUE TITLE
[Feature] Create small .js file conaining ONLY search-related functionality (for use in browser)

### ISSUE THREAD

**Issue by stargazer33** [typesense/typesense-js#222]
## Description
Having full functionality of Typeset API in one file (typesense.min.js, about 95 kB) is great - when you need this functionality.
But when your code works in web browser - all you do is something like this:
```
   typesense = new Typesense.SearchClient(...)
   ...
   typesense.collections('test').documents().search({...}).then(
        function (searchResults){
            ...
            searchResults.hits
            ...
        }       
```
I mean - the end-user is not creating collections/documents/synonyms/aliases...
The end-user just searching.

## Expected Behavior
There is a need for something like  `typesense-search.js`.
It should contains Typesense.SearchClient and ... basically that's it.
Should have enough functionality to do search in browser.
Should be... I would say... less than 30 kB

**Comment by jasonbosco** [comment]
There's a `SearchClient` class that does exactly this: https://github.com/typesense/typesense-js/blob/master/src/Typesense/SearchClient.ts

If you import that class (instead of `Client), your bundler should automatically tree-shake and only include a smaller subset of the files just required for search in your final build.

**Comment by stargazer33** [comment]
> If you import that class (instead of `Client) ...

OK, I did it:
```
  "dependencies": {
    ...
    "typesense": "1.8.2"
  },
```

```
import { SearchClient as TypesenseSearchClient } from "typesense";
const typesense = new TypesenseSearchClient({
....
```

```
npm ls --depth 1
...
└─┬ typesense@1.8.2
  ├── @babel/runtime@7.25.0 deduped
  ├── axios@1.7.4
  └── loglevel@1.9.1

```

`parcel build --dist-dir ./dist --reporter @parcel/reporter-bundle-analyzer`
And analyzer tells us this:
[axios](https://github.com/user-attachments/assets/3a0356c0-d1d0-479d-bee7-76a8968569b7)  36.8 KB
[typesense](https://github.com/user-attachments/assets/d4da6b26-7483-4c82-9826-fcc0b0f9ea06)  32.3 KB
 
 
 69 KB altogether.
 Not a small thing.

(correct me if i wrong)

**Comment by stargazer33** [comment]
UPDATE:

Running `parcel build --log-level verbose` I got a long list of messages like this:
```
...
@parcel/transformer-js: Unknown usage of CommonJS `exports` object. This causes tree shaking to be disabled.
  ...node_modules/typesense/lib/Typesense/Preset.js:38:24
    37 | };
  > 38 | var __importDefault = (this && this.__importDefault) || function (mod) {
  >    |                        ^
    39 |     return (mod && mod.__esModule) ? mod : { "default": mod };
    40 | };
...
@parcel/transformer-js: Conditional or non-top-level `require()` call. This causes the resolved module and all dependencies to be wrapped.
  ...node_modules/typesense/lib/Typesense/Client.js:22:1
    21 | var Preset_1 = __importDefault(require("./Preset"));
  > 22 | var Analytics_1 = __importDefault(require("./Analytics"));
  >    | ^
    23 | var Stopwords_1 = __importDefault(require("./Stopwords"));
    24 | var Stopword_1 = __importDefault(require("./Stopword"));
...    
```

About 10 screens of such messages.
Well, it could be that something should be improved in code... and after this - tree shaking will work better...

**Comment by boldtrn** [comment]
Yes, I fully agree, the bundle size should really be optimized. For browser use I am not even sure we need axios at all? Given that nodejs has a ntive fetch implementation by now, IMHO this should be completely removed.

**Comment by tharropoulos** [comment]
> Yes, I fully agree, the bundle size should really be optimized. For browser use I am not even sure we need axios at all? Given that nodejs has a ntive fetch implementation by now, IMHO this should be completely removed.

We support Node versions from 18 onwards. `fetch` was introduced as an experimental feature in Node 18, and wasn't considered stable until Node 21. This means that in order to support pure `fetch` as opposed to axios (which does have a large bundle size sadly), we would need to deprecate versions of node < 22 (22 being the latest LTS).

In terms of bundle sizes and tree-shaking, I've posted a PR #261 to migrate the build system over to esbuild, and refactored the sourcepoint `src/Typesense.ts` to not use default exports, in order to avoid bringing in unnecessary dependencies alongside the `SearchClient` class. I tested it on both Node (18 and 23), a Vite React project and a Parcel project. 

* On Vite with rollup, I saw a 100kB difference in size flat, plus a 15kB extra size difference when using the SearchClient.
* On Parcel, there was a 80kB difference in size, and an extra 12kB size difference when using the SearchClient

This migration offers a cleaner access to the consumer API as well, as all types are exported by default on the entrypoint, without needing to go into `/lib` anymore.

**Comment by stargazer33** [comment]
> On Vite with rollup, I saw a 100kB difference in size flat, plus a 15kB extra size difference when using the SearchClient.

Looks promising. Yeah.
Would like to test this minimized typesense, let us know in which version is will be available.

By the way, I will switch to Astro for all frontend things, and Astro has Vite built-in, so will test with Vite...

**Comment by tharropoulos** [comment]
>  Would like to test this minimized typesense, let us know in which version is will be available.

Most of the bundle size comes from Axios itself. The library itself without the dependencies takes about 15kB in size

![Image](https://github.com/user-attachments/assets/2065c824-7ebd-479d-93c5-e3cb7352e905)

**Comment by boldtrn** [comment]
Since node 18 is EOL, how about dropping node 18 support and dropping axios? That should fix the size issue.
